.github/workflows/yki.yml

```diff
@@ -0,0 +1,70 @@
+name: YKI
+
+on:
+  workflow_dispatch:
+  push:
+    paths-ignore:
+      - '**.md'
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Prepare Clojure env
+        uses: ./.github/actions/prepare-clojure-env
+
+      - name: Run tests
+        run: lein test
+
+  build:
+    runs-on: ubuntu-latest
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Prepare Clojure env
+        uses: ./.github/actions/prepare-clojure-env
+
+      - name: Build uberjar
+        run: lein uberjar
+
+      - uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: target
+          key: ${{ github.sha }}
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: [test, build]
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: target
+          key: ${{ github.sha }}
+
+      - name: Build and deploy Docker image
+        shell: bash
+        run: |
+          git clone https://github.com/Opetushallitus/ci-tools.git
+          source ci-tools/common/setup-tools.sh
+          mv target/yki-*-standalone.jar $DOCKER_BUILD_DIR/artifact/yki.jar
+          cp -vr oph-configuration $DOCKER_BUILD_DIR/config/
+          export BASE_IMAGE="baseimage-fatjar-openjdk17:master"
```

Comment:
Aikaisemmin ollut käytössä `baseimage-fatjar-openjdk11:master`

Reply:
Tätä mietin kun vertasin .travis.yml:ään, että olisko parempi pitäytyä nykyisessä openjdk11 imagessa?

Reply:
Noo, ehkä. Hyvin stabiilihan tuo 17 on myös ollut ja mielellään siihen päivittäisi. Voi sen toki sitten erillisenäkin taskina päivittää.

Reply:
Tähän määrittää nyt tietysti Java version, joka käytössä kun softaa ajetaan? Ilmeisesti ei pitäisi olla ongelmaa vaihtaa versiosta suurempaan esim. 11 -> 17: https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html

Tältä puolin varmaan ok tehdä mut hyvä ois testata untuvalla/QA:lla että kaikki vaikuttaa muutoksen jälkeen toimivan?

Reply:
Joo, juuri näin. Clojureakin on aika monta vuotta tullut ajettua JDK17 päällä eikä siinä *pitäisi* mitään ongelmia olla. Toki untuvalla ja QA:lla testaukset kannattaa ihan varmuuden vuoksi ensin tehdä.

Reply:
Joo, eiköhän tää muutos oo sitten varsin ok. Lokaalisti oon ajanut jdk11:lla kun jenv käytössä

Reply:
Untuvalla nyt ajossa JDK17 päällä.